cloudflare/resource_cloudflare_load_balancer_pool.go

```diff
@@ -242,7 +291,7 @@ func resourceCloudflareLoadBalancerPoolRead(d *schema.ResourceData, meta interfa
 	d.Set("created_on", loadBalancerPool.CreatedOn.Format(time.RFC3339Nano))
 	d.Set("modified_on", loadBalancerPool.ModifiedOn.Format(time.RFC3339Nano))
 
-	if err := d.Set("origins", flattenLoadBalancerOrigins(loadBalancerPool.Origins)); err != nil {
+	if err := d.Set("origins", flattenLoadBalancerOrigins(d, loadBalancerPool.Origins)); err != nil {
```

Comment:
looks like the `headers` values are not getting persisted to state when running the tests. i haven't dug too deep but if you make the other fixes and run the `make testacc` command targetting the new load balancer tests, you get the following.

```
=== CONT  TestAccCloudflareLoadBalancerPool_FullySpecified
    testing.go:705: Step 0 error: After applying this step and refreshing, the plan was not empty:

        DIFF:

        UPDATE: cloudflare_load_balancer_pool.xxeocwmfcj
          check_regions.#:             "1" => "1"
          check_regions.0:             "WEU" => "WEU"
          created_on:                  "2021-04-25T23:58:19.796021Z" => "2021-04-25T23:58:19.796021Z"
          description:                 "tfacc-fully-specified" => "tfacc-fully-specified"
          enabled:                     "false" => "false"
          id:                          "6be5cef5434f66dd1b50dd755b81931c" => "6be5cef5434f66dd1b50dd755b81931c"
          minimum_origins:             "2" => "2"
          modified_on:                 "2021-04-25T23:58:19.796021Z" => "2021-04-25T23:58:19.796021Z"
          monitor:                     "" => ""
          name:                        "my-tf-pool-basic-xxeocwmfcj" => "my-tf-pool-basic-xxeocwmfcj"
          notification_email:          "someone@example.com" => "someone@example.com"
          origins.#:                   "2" => "2"
          origins.0.address:           "192.0.2.1" => "192.0.2.1"
          origins.0.enabled:           "false" => "false"
          origins.0.header.#:          "0" => "1"
          origins.0.header.0.header:   "" => "Host"
          origins.0.header.0.values.#: "" => "1"
          origins.0.header.0.values.0: "" => "terraform.cfapi.net"
          origins.0.name:              "example-1" => "example-1"
          origins.0.weight:            "1" => "1"
          origins.1.address:           "192.0.2.2" => "192.0.2.2"
          origins.1.enabled:           "true" => "true"
          origins.1.header.#:          "0" => "1"
          origins.1.header.0.header:   "" => "Host"
          origins.1.header.0.values.#: "" => "1"
          origins.1.header.0.values.0: "" => "terraform2.cfapi.net"
          origins.1.name:              "example-2" => "example-2"
          origins.1.weight:            "0.5" => "0.5"



        STATE:

        cloudflare_load_balancer_pool.xxeocwmfcj:
          ID = 6be5cef5434f66dd1b50dd755b81931c
          provider = provider.cloudflare
          check_regions.# = 1
          check_regions.0 = WEU
          created_on = 2021-04-25T23:58:19.796021Z
          description = tfacc-fully-specified
          enabled = false
          minimum_origins = 2
          modified_on = 2021-04-25T23:58:19.796021Z
          monitor =
          name = my-tf-pool-basic-xxeocwmfcj
          notification_email = someone@example.com
          origins.# = 2
          origins.0.address = 192.0.2.1
          origins.0.enabled = false
          origins.0.name = example-1
          origins.0.weight = 1
          origins.1.address = 192.0.2.2
          origins.1.enabled = true
          origins.1.name = example-2
          origins.1.weight = 0.5
--- FAIL: TestAccCloudflareLoadBalancerPool_FullySpecified (2.66s)
```

Reply:
I updated the PR, but I can't run that test, what can be wrong?

```TF_LOG=DEBUG TESTARGS="--run TestAccCloudflareLoadBalancerPool_FullySpecified -count=1" make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./...) -v --run TestAccCloudflareLoadBalancerPool_FullySpecified -count=1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareLoadBalancerPool_FullySpecified
=== PAUSE TestAccCloudflareLoadBalancerPool_FullySpecified
=== CONT  TestAccCloudflareLoadBalancerPool_FullySpecified
2021/04/27 23:57:35 [DEBUG] Test: Executing step 0
    testing.go:705: Step 0 error: Missing newline after argument: On /var/folders/jm/nl012fkn62l7vdp94w7rwf080000gn/T/tf-test787253313/main.tf line 11: An argument definition must end with a newline.
2021/04/27 23:57:35 [WARN] Skipping destroy test since there is no state.
--- FAIL: TestAccCloudflareLoadBalancerPool_FullySpecified (0.00s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	0.567s
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
FAIL
make: *** [testacc] Error 1
```

Also, I think the issue with the persisted state can be because we change the header in the schema, but don't change it in the test, can we?
Could you please re-run it with my latest changes?

Reply:
I a little bit updated the test and catch the same issue.

Reply:
i'm seeing a different exception now

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareLoadBalancerPool" -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareLoadBalancerPool_Import
=== PAUSE TestAccCloudflareLoadBalancerPool_Import
=== RUN   TestAccCloudflareLoadBalancerPool_Basic
=== PAUSE TestAccCloudflareLoadBalancerPool_Basic
=== RUN   TestAccCloudflareLoadBalancerPool_FullySpecified
=== PAUSE TestAccCloudflareLoadBalancerPool_FullySpecified
=== RUN   TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy
=== PAUSE TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy
=== CONT  TestAccCloudflareLoadBalancerPool_Import
--- PASS: TestAccCloudflareLoadBalancerPool_Import (2.73s)
=== CONT  TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy
--- PASS: TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy (5.02s)
=== CONT  TestAccCloudflareLoadBalancerPool_FullySpecified
    testing.go:705: Step 0 error: Check failed: Check 6/6 error: cloudflare_load_balancer_pool.oaecdixxec: Attribute 'origins.0.header.0.header' not found
--- FAIL: TestAccCloudflareLoadBalancerPool_FullySpecified (2.25s)
=== CONT  TestAccCloudflareLoadBalancerPool_Basic
--- PASS: TestAccCloudflareLoadBalancerPool_Basic (3.07s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	13.450s
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
FAIL
make: *** [testacc] Error 1
```

I suspect it is due to `origins` being a `TypeSet` and index position "0" doesn't exist due to the index being a hash value (see "state representation at https://www.terraform.io/docs/extend/schemas/schema-types.html#typeset).